switools/signaturelib.py

```diff
@@ -12,4 +13,23 @@ def getSigFileName( swiFile ):
       return SWIX_SIG_FILE_NAME
    return SWI_SIG_FILE_NAME
 
+def getOptimizations( swi, workDir ):
+   ret = os.system( "set -e; swi=$(readlink -f %s); cd %s; "
+                    "unzip -q -o $swi swimSqshMap" % ( swi, workDir ) )
+   if ret:
+      return None # legacy image
+   optims = []
+   with open( "%s/swimSqshMap" % workDir ) as f:
+      for line in f:
+         optim, _ = line.split( "=", 1 )
+         optims.append( optim )
+   os.system( "rm %s/swimSqshMap" % workDir )
+   return optims
 
+def extractSwadapt( swi, workDir ):
+   ret = os.system( "set -e; image=$(readlink -f %s); cd %s;"
+                    "unzip -o -q $image swadapt" % ( swi, workDir ) )
+   if ret:
+      print( "Error: '%s' does not contain the 'swadapt' utility" % swi )
+      shutil.rmtree( workDir )
```

Comment:
uhm, indeed, that's nicer, especially given unzip's nasty willingness to abide by its own -qq argument (very quiet) and printing warnings when the file to extract does not exist....

Reply:
`zipfile` is so much nicer to use than `unzip`. One cool thing I was able to do with it was to mount squashfs files contained in a swix without extracting them.  You can't do that with `unzip` 😃

Reply:
`zipfile` doesn't preserve permissions when extracting so this needs to be added after `swi.extract`:
```
os.chmod('{}/swadapt'.format(workDir), 0o755)
```

Reply:
So, while `zipfile` is nice to use and makes some things very easy, it has a couple of bugs which make it unsuitable for some of the things we need to do in these scripts, so I'll try to mark those suggestions as resolved.  It might still be useful in some places (e.g. it's easier to check if a file is in the archive with `zipfile`), but not in others.